indico/modules/search/client/js/components/SideBar.jsx

```diff
@@ -6,130 +6,84 @@
-  const [affiliationFilter, setAffiliationFilter] = useQueryParam('affiliation', StringParam);
-
-  const addSearchFilter = (value, type) => {
-    if (type === 'speaker') {
-      setSpeakerFilter(speakerValue === value ? '' : value, 'pushIn');
-      setSpeakerValue(speakerValue === value ? '' : value);
-    } else if (type === 'affiliation') {
-      setAffiliationFilter(affiliationValue === value ? '' : value, 'pushIn');
-      setAffiliationValue(affiliationValue === value ? '' : value);
-    }
-  };
-
-  const handleChange = (e, {value}) => {
-    addSearchFilter(value.name, value.type);
+export default function SideBar({query, aggregations, onChange}) {
+  const handleChange = (type, value) => {
+    onChange && onChange(type in query && query[type] === value ? undefined : value, type);
   };
 
-  // it's going to change soon
-  if (filterType !== 'Contributions') {
-    return;
-  }
   return (
     <div styleName="sidebar">
-      <Table celled padded>
-        <Table.Header>
-          <Table.Row>
-            <Table.HeaderCell>Speakers</Table.HeaderCell>
-          </Table.Row>
-        </Table.Header>
-
-        <Table.Body>
-          <Table.Row>
-            <Table.Cell>
-              <Checkbox
-                label="Aristofanis (45)"
-                name="Aristofanis"
-                value={{name: 'Aristofanis', type: 'speaker'}}
-                checked={speakerValue === 'Aristofanis'}
-                onChange={handleChange}
-              />
-            </Table.Cell>
-          </Table.Row>
-          <Table.Row>
-            <Table.Cell>
-              <Checkbox
-                label="Pedro (4555)"
-                name="Pedro"
-                value={{name: 'Pedro', type: 'speaker'}}
-                checked={speakerValue === 'Pedro'}
-                onChange={handleChange}
-              />
-            </Table.Cell>
-          </Table.Row>
-          <Table.Row>
-            <Table.Cell>
-              <Checkbox
-                label="Adrian (455)"
-                name="Adrian"
-                value={{name: 'Adrian', type: 'speaker'}}
-                checked={speakerValue === 'Adrian'}
-                onChange={handleChange}
-              />
-            </Table.Cell>
-          </Table.Row>
-        </Table.Body>
-      </Table>
+      {aggregations
+        .filter(({buckets}) => buckets && buckets.length)
+        .map(({key, label, buckets}) => (
+          <AggregationList
+            key={key}
+            name={key}
+            title={label}
+            items={buckets}
+            query={query}
+            onChange={handleChange}
+          />
+        ))}
+    </div>
+  );
+}
 
-      <Table celled padded>
-        <Table.Header>
-          <Table.Row>
-            <Table.HeaderCell>Affiliations</Table.HeaderCell>
-          </Table.Row>
-        </Table.Header>
+function AggregationList({name, title, items, query, onChange}) {
+  const _items = items
```

Comment:
IMHO `_items` vs `items` gets a bit confusing... what about just reassigning `items = items.map(...)` instead?

Reply:
I agree, it seems confusing.